File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/go/exercises/practice/simple-linked-list/simple_linked_list.go
package linkedlist

import "errors"

type Element struct {
	value int
	next  *Element
}

type List struct {
	head *Element
	size int
}

func New(elements []int) *List {
	l := &List{}
	// Add elements in normal order
	for i := 0; i < len(elements); i++ {
		l.Push(elements[i])
	}
	return l
}

func (l *List) Size() int {
	return l.size
}

func (l *List) Push(element int) {
	newElem := &Element{value: element}
	if l.head == nil {
		l.head = newElem
	} else {
		current := l.head
		for current.next != nil {
			current = current.next
		}
		current.next = newElem
	}
	l.size++
}

func (l *List) Pop() (int, error) {
	if l.head == nil {
		return 0, errors.New("list is empty")
	}

	// Handle single element case
	if l.head.next == nil {
		value := l.head.value
		l.head = nil
		l.size--
		return value, nil
	}

	// Find second-to-last element
	previous := l.head
	current := l.head.next
	for current.next != nil {
		previous = current
		current = current.next
	}

	value := current.value
	previous.next = nil
	l.size--
	return value, nil
}

func (l *List) Array() []int {
	result := []int{}
	current := l.head
	for current != nil {
		result = append(result, current.value)
		current = current.next
	}
	return result
}

func (l *List) Reverse() *List {
	current := l.head
	var reversedHead *Element
	for current != nil {
		next := current.next
		current.next = reversedHead
		reversedHead = current
		current = next
	}
	return &List{head: reversedHead, size: l.size}
}

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/simple-linked-list/simple_linked_list.py
class EmptyListException(Exception):
    """Exception raised when the linked list is empty."""
    def __init__(self, message):
        super().__init__(message)
        self.message = message


class Node:
    def __init__(self, value):
        self._value = value
        self._next = None

    def value(self):
        return self._value

    def next(self):
        return self._next


class LinkedList:
    def __init__(self, values=None):
        self._head = None
        self._length = 0
        if values is not None:
            for value in values:
                self.push(value)

    def __iter__(self):
        current = self._head
        while current is not None:
            yield current.value()
            current = current.next()

    def __len__(self):
        return self._length

    def head(self):
        if self._head is None:
            raise EmptyListException("The list is empty.")
        return self._head

    def push(self, value):
        new_node = Node(value)
        new_node._next = self._head
        self._head = new_node
        self._length += 1

    def pop(self):
        if self._head is None:
            raise EmptyListException("The list is empty.")
        value = self._head.value()
        self._head = self._head.next()
        self._length -= 1
        return value

    def reversed(self):
        reversed_list = LinkedList()
        current = self._head
        while current is not None:
            reversed_list.push(current.value())
            current = current.next()
        return reversed_list
